docs/how_to_guides/how_to_use_apparent_and_true_chemical_species.rst

```diff
@@ -0,0 +1,213 @@
+How to use apparent and true chemical species
+=============================================
+
+In ProteusLib, most all chemical processes simulated will be considered "true"
+species, i.e., species that actually exist in an aqueous solution. However, their
+may be times when you want to have your system report the "apparent" species
+between your inlet and/or outlet ports of a flowsheet. To do this, you need
+to understand how to define a species in a given configuration file as "apparent",
+which is a special component type in the **thermo-properties** configuration dictionary.
+
+For more information on creating a **thermo-properties** configuration dictionary,
+see 'How to setup simple chemistry'.
+
+
+What you need to update in the thermo-properties configuration dictionary
+-------------------------------------------------------------------------
+
+1. Add the ``"Apparent"`` and ``"StateIndex"`` objects under your import statements
+2. Specify specific species as ``"Apparent"`` within its component definition
+3. Give a list of "true" species it will dissociate into
+4. Define a "state_components" argument in the config and specify ``"StateIndex.true"``
+
+
+Example Configuration Dictionary
+--------------------------------
+
+This is best understood by going through an example. In this case, we will consider
+building off of the 'How to setup simple chemistry' guide and add ``"NaCl"`` as
+an ``"Apparent"`` species.
+
+.. testsetup::
+
+   # quiet idaes logs
+   import idaes.logger as idaeslogger
+   idaeslogger.getLogger('ideas.core').setLevel('CRITICAL')
+   idaeslogger.getLogger('idaes.init').setLevel('CRITICAL')
+
+.. testcode::
+
+  from pyomo.environ import units as pyunits
+  from idaes.core import AqueousPhase
+
+  # Add Apparent in this import statement
+  from idaes.core.components import Solvent, Cation, Anion, Apparent
+
+  import idaes.generic_models.properties.core.pure.Perrys as Perrys
+  from idaes.generic_models.properties.core.pure.ConstantProperties import Constant
+  from idaes.generic_models.properties.core.state_definitions import FTPx
+  from idaes.generic_models.properties.core.eos.ideal import Ideal
+
+  # Add the import for StateIndex
+  from idaes.generic_models.properties.core.generic.generic_property import StateIndex
+
+  # Configuration dictionary
+  thermo_config = {
+      "components": {
+          'H2O': {"type": Solvent,
+                # Define the methods used to calculate the following properties
+                "dens_mol_liq_comp": Perrys,
+                "enth_mol_liq_comp": Perrys,
+                "cp_mol_liq_comp": Perrys,
+                "entr_mol_liq_comp": Perrys,
+                # Parameter data is always associated with the methods defined above
+                "parameter_data": {
+                      "mw": (18.0153, pyunits.g/pyunits.mol),
+                      # Parameters here come from Perry's Handbook:  p. 2-98
+                      "dens_mol_liq_comp_coeff": {
```

Comment:
Now that I think about it, is there a way for the user to avoid having to specify these parameter data from Perry's Handbook in this example? Is that where the EDB will come into play? I was just thinking that if a user didn't have these parameter values readily available, it could be painful... but I am assuming that the database (or some other means) would be used to avoid manually entering such values (at least for common configurations).

Reply:
On the other hand, this is useful because it shows the user how to manually enter such parameters.

Reply:
@adam-a-a  Yes, eventually much of this will be done by EDB. EDB is in parallel development with the rest of the chem team, so for the time being I am doing this stuff manually. Future documentation for how to do this with EDB is going to come (#140).

Reply:
We need to be careful with this however - we need to make sure we have the right to reproduce data like this. Also, the IDAES approach has generally be to provide a way for users to input data, but to leave finding the data to the user.

Reply:
Sounds to me that, at least for the purpose of this example, we should just use any open source parameters or methods. The exact values of parameters are not important for any of these docs.